jsoc/gsoc/medpipe3d.md

```diff
@@ -0,0 +1,82 @@
+
+- **Mentor:** Jakub Mitura [email: jakub.mitura14@gmail.com]
+
+- **Difficulty**: Hard
+
+- **Duration**: 350 hours
+
+- **Suggested Skills and Background**: 
+  - Experience with Julia
+  - Basic familiarity with computer graphics preferably OpenGL 
+  - Some experience with 3d volumetric data with spatial metadata (or a willingness to learn!)
+
+- **Potential Outcomes:** 
+Although MedEye3D already supports displaying medical images, there are still some functionalities that will be useful for the implementation of some more advanced algorithms, like supervoxel segmentation or image registration (and both of them are crucial for solving a lot of important problems in medical imaging). To achieve this this project's goal is to implement.
+1) Developing support for multiple image viewing with indicators for image registration like display of the borders, and display lines connecting points.
+2) Automatic correct windowing for MRI and PET.
+3) Support of display for supervoxels (sv). Show borders of sv; indicate whether the gradient of the image is in agreement with sv borders.
+4) Improve start time.
+5) Simplify basic usage by providing high-level functions.
+
+- **Success criteria and time needed:** How the success of functionality described above is defined and the approximate time required for each.
+
+1) The user can load 2 different images, and they would display concurrently one next to the other. During scrolling the same area of the body should be displayed (for well-registered sample images) based on the supplied metadata. While moving the mouse cursor on one image the position of the cursor in the same physical spot on the other image should be displayed (physical location calculated from spatial metadata).  120h
+2) Given the most common PET and MRI modalities (random FDG PET/CT, and T2, T1, FLAIR, ADC, DWI on MRI) - the user will see the image similar to what is automatically displayed in 3DSlicer - 10h
+3) Given an integer mask where a unique integer value will encode information about a single supervoxel and an underlying 3d medical image user will have the option to overlay the original image with the borders of the superpixels where adjacent borders will have different colors, or show those borders on the background of the image convolved with edge filter, for example, Sobel filter - 180h
+4) Any measurable decrease in the start time of the viewer -   20h
+5) The user will be able to display images just by supplying MedImage objects from the MedImage.jl library to a single display function -  20h
+
+
+
+
+### Project 2: Adding dataset-wide functions and integrations of augmentations
+
+**Description:** 
+MedPipe3D was created as a package that improves integration between other parts of the small ecosystem (MedEye3D, MedEval3D, and MedImage). Currently, it needs to be expanded and adapted so it can be a basis for a fully functional medical imaging pipeline.

```

Comment:
Could you expand this description further to say in what ways it needs to be expanded/adapted?